pkg/pool-utils/test/PoolRegistrationLib.test.ts

```diff
@@ -112,6 +118,85 @@ describe('PoolRegistrationLib', function () {
     });
   });
 
+  describe('registerComposablePool', () => {
+    let assetManagers: string[];
+
+    sharedBeforeEach(async () => {
+      assetManagers = Array.from({ length: tokens.length }, () => ethers.Wallet.createRandom().address);
+    });
+
+    it('registers the pool in the vault', async () => {
+      const poolId = await registerComposablePool(PoolSpecialization.GeneralPool, assetManagers);
+
+      const { address: poolAddress } = await vault.getPool(poolId);
+      expect(poolAddress).to.equal(lib.address);
+    });
+
+    it('registers the pool with the correct specialization', async () => {
+      // Composable pools cannot use the TwoTokenPool specialization as they must have at least 3 tokens.
+      const specializations: (keyof typeof PoolSpecialization)[] = ['GeneralPool', 'MinimalSwapInfoPool'];
+      for (const specialization of specializations) {
+        const poolId = await registerPool(PoolSpecialization[specialization]);
+        const { specialization: poolSpecialization } = await vault.getPool(poolId);
+        expect(poolSpecialization).to.equal(PoolSpecialization[specialization]);
+      }
+    });
+
+    it('registers the tokens correctly', async () => {
+      const poolId = await registerComposablePool(PoolSpecialization.GeneralPool, assetManagers);
+      const { tokens: poolTokens } = await vault.getPoolTokens(poolId);
+
+      // The library mock is fulfilling the role of the Pool in this test.
+      const composableTokens = [lib.address, ...tokens.addresses];
+
+      expect(poolTokens).to.deep.eq(composableTokens);
+    });
+
+    it('reverts if the tokens are not sorted', async () => {
```

Comment:
Also add a test that checks we revert if one of the tokens is the Pool (library)